pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -57,11 +60,48 @@ func newPodCache(c *Controller, informer coreinformers.PodInformer, queueEndpoin
 		IPByPods:           make(map[string]string),
 		needResync:         make(map[string]sets.Set),
 		queueEndpointEvent: queueEndpointEvent,
+		podTransitionTime:  make(map[string]metav1.Time),
```

Comment:
Is this to record the old ready status and compare with the current ready status? If so, we donot need the record the transition time.

Reply:
If we don't use transition time, we would still need a flag to store the pod status. IMO, it's probably better to use the time because that will help with debugging when issue occurs.